plugins/filesystem/local/adapter/adapter.php

```diff
@@ -12,6 +12,9 @@
 JLoader::import('joomla.filesystem.file');
 JLoader::import('joomla.filesystem.folder');
 
+require_once 'filenotfoundexception.php';
```

Comment:
But the api model needs to know what it gets when it asks for an adapter, so the interface needs to stay there. com_media should not have any dependency to a plugin but the plugins to com_media.